src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
+            which means that the threshold is exceeded if the forecast is greater than or equal
+            to the threshold. Alternatively, you can provide ``threshold_mode='>'` which means
+            that the threshold is exceeded if the forecast is strictly greater than the threshold.
+
+
+    Returns:
+        The Brier score for the ensemble forecast.
+
+    Raises:
+        ValueError: if values in ``thresholds`` are not monotonically increasing.
+        ValueError: if ``fcst`` contains the dimension 'threshold'.
+        ValueError: if ``obs`` contains the dimension 'threshold'.
+        ValueError: if ``weights`` contains the dimension 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is not in ``fcst`` dimensions.
+
+    References:
+        - Ferro, C. A. T. (2013). Fair scores for ensemble forecasts. Quarterly
+            Journal of the Royal Meteorological Society, 140(683), 1917–1923.
+            https://doi.org/10.1002/qj.2270
+
+    See Also:
+        - :py:func:`scores.probability.brier_score`
+
+    Examples:
+        Calculate the Brier score for an ensemble forecast for a single threshold:
+
+        >>> import numpy as np
+        >>> import xarray as xr
+        >>> from scores.probability import ensemble_brier_score
+        >>> fcst = xr.DataArray(np.random.rand(10, 10), dims=['time', 'ensemble'])
+        >>> obs = xr.DataArray(np.random.rand(10), dims=['time'])
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=0.5)
+
+        Calculate the Brier score for an ensemble forecast for multiple thresholds:
+        >>> thresholds = [0.1, 0.5, 0.9]
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=thresholds)
+
+    """
+    if threshold_mode not in [">=", ">"]:
+        raise ValueError("threshold_mode must be either '>=' or '>'.")
+    if ensemble_member_dim == "threshold":
+        raise ValueError("The ensemble_member_dim is not allowed to be 'threshold'.")
+    if "threshold" in fcst.dims:
+        raise ValueError("The dimension 'threshold' is not allowed in fcst.")
+    if "threshold" in obs.dims:
+        raise ValueError("The dimension 'threshold' is not allowed in obs.")
+    if ensemble_member_dim not in fcst.dims:
+        raise ValueError("ensemble_member_dim must be one of the dimensions in fcst.")
+
+    weights_dims = None
+    if weights is not None:
+        if "threshold" in weights.dims:
+            raise ValueError("The dimension 'threshold' is not allowed in weights.")
+        weights_dims = weights.dims
+
+    dims_for_mean = gather_dimensions(
+        fcst.dims,
+        obs.dims,
+        weights_dims=weights_dims,
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+        score_specific_fcst_dims=ensemble_member_dim,
+    )
+    if isinstance(thresholds, (float, int)):
+        thresholds = [thresholds]
```

Comment:
If this is a general pattern for ensemble-handling functions, it could perhaps be turned into a general function for processing threshold parameters and used consistently across ensemble-handling functions to ensure consistent and bug-free code. Up to you.

Reply:
I don't think that this pattern is used anywhere else